src/components/Courses/Courses.jsx

```diff
@@ -1,34 +1,53 @@
-import { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
-
+import { useSelector, useDispatch } from "react-redux";
+import { getCoursesService } from "../../services";
 import SearchBar from "./components/SearchBar/SearchBar";
 import CourseCard from "./components/CourseCard/CourseCard";
 import Button from "../../common/Button/Button";
-
-import { mockedCoursesList } from "../../constants";
-
 import styles from "./Courses.module.css";
+import { PATH_URIS } from "../../constants";
+import { BUTTON_TEXT, LOG_MESSAGES } from "./coursesStrings";
 
 function Courses() {
   const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const courses = useSelector((state) => state.courses);
```

Comment:
ideally this would be a name selector re-used throughout the app, like "useSelector(selectCourses)`. Then it becomes kind of like a constant which is easier to refactor if we decide to change the store's hierarchy

Reply:
done, included as a constant in a new class named selectors